### ISSUE TITLE
Phase 1 Step 2: Configuration Management - Comprehensive Implementation

### ISSUE THREAD

**Issue by dhar174** [dhar174/steel_defect_demo#19]
## Overview

Implement a comprehensive configuration management system for the steel defect prediction system based on the detailed specifications in `TECHNICAL_IMPLEMENTATION_SPEC.md`. This system will manage all configuration aspects for data generation, model training, and inference operations.

### 1. Data Generation Configuration (`configs/data_generation.yaml`)

**Primary Features:**
- **Sensor Parameter Configuration:**
  - Casting Speed: base_value: 1.2 m/min, noise_std: 0.05, range: 0.8-1.8 m/min
  - Mold Temperature: base_value: 1520°C, noise_std: 10, range: 1480-1580°C
  - Mold Level: base_value: 150mm, noise_std: 5, range: 120-180mm
  - Cooling Water Flow: base_value: 200 L/min, noise_std: 15, range: 150-250 L/min
  - Superheat: base_value: 25°C above liquidus, noise_std: 3, range: 15-40°C

- **Defect Simulation Parameters:**
  - Defect probability: 0.15 (15% of casts)
  - Trigger conditions:
    - Prolonged mold level deviation: 30 seconds
    - Rapid temperature drop: 50°C in 60 seconds
    - High speed with low superheat combination
  
- **Output Format Specifications:**
  - Raw data format: Parquet files
  - Metadata format: JSON files
  - Train/test split ratio: 0.8/0.2

- **Data Generation Parameters:**
  - Number of casts: 1200
  - Cast duration: 120 minutes each
  - Sampling rate: 1 Hz
  - Random seed: 42 for reproducibility

### 2. Model Configuration (`configs/model_config.yaml`)

**Baseline XGBoost Model:**
- Algorithm parameters: n_estimators=100, max_depth=6, learning_rate=0.1
- Feature engineering flags: statistical, stability, duration, interaction features
- Validation: 5-fold CV, early stopping, AUC evaluation metric

**LSTM Model Architecture:**
- Input size: 5 sensors
- Hidden size: 64 units
- Layers: 2 stacked LSTM layers
- Dropout: 0.2 for regularization
- Training: batch_size=32, learning_rate=0.001, 100 epochs max
- Sequence processing: 300-second windows (5 minutes at 1Hz)

**Evaluation Metrics:**
- AUC-ROC, AUC-PR, F1-score, precision, recall, accuracy
- Test size: 0.2, stratified sampling

### 3. Inference Configuration (`configs/inference_config.yaml`)

**Real-time Simulation:**
- Playback speed: 10x real-time
- Update interval: 30 seconds
- Buffer size: 300 seconds (5 minutes)

**Decision Thresholds:**
- Defect probability threshold: 0.5
- High risk threshold: 0.7
- Alert threshold: 0.8

**Dashboard and Monitoring:**
- Dashboard enabled on port 8050
- Prediction logging and trajectory saving
- Performance tracking and data drift detection

### Configuration Loading Utility

Create `src/utils/config_loader.py` with the following capabilities:
- YAML configuration file parsing
- Environment variable override support
- Configuration validation and schema checking
- Default value handling
- Configuration merging for different environments

### Integration Points

The configuration system must integrate with:
1. **Data Generator** (`src/data/data_generator.py`) - sensor parameters and simulation rules
2. **Feature Engineering** (`src/features/feature_engineer.py`) - processing parameters
3. **Model Training** (`scripts/train_baseline_model.py`, `scripts/train_lstm_model.py`) - hyperparameters
4. **Inference Engine** (`src/inference/inference_engine.py`) - thresholds and processing settings
5. **Dashboard** (`src/visualization/dashboard.py`) - display and monitoring parameters

### Validation Requirements

- Schema validation for all configuration files
- Range checking for numerical parameters
- Dependency validation (e.g., sequence_length compatibility with sampling_rate)
- Configuration compatibility checks between components

### Environment Support

- Development, testing, and production configuration variants
- Environment-specific overrides capability
- Secure handling of sensitive parameters (future database connections, API keys)

### File Structure
```
configs/
├── data_generation.yaml
├── model_config.yaml
├── inference_config.yaml
└── schemas/
    ├── data_generation_schema.yaml
    ├── model_config_schema.yaml
    └── inference_config_schema.yaml
```

### Dependencies
- `pyyaml>=6.0` for YAML parsing
- `hydra-core>=1.2.0` for advanced configuration management (optional)
- `pydantic>=1.10.0` for configuration validation

### Testing Requirements
- Unit tests for configuration loading and validation
- Integration tests with dependent components
- Configuration schema validation tests
- Environment override functionality tests

## Acceptance Criteria

- [ ] All three configuration files implemented with complete parameter sets
- [ ] Configuration loader utility with validation and error handling
- [ ] Schema validation for all configuration files
- [ ] Integration with all dependent components (data generation, training, inference)
- [ ] Environment-specific configuration support
- [ ] Comprehensive unit and integration tests
- [ ] Documentation for configuration options and usage
- [ ] Example configurations for different use cases

## Dependencies

This issue builds upon:
- Project directory structure setup
- Basic Python environment and dependencies
- Initial implementation of dependent modules (data generator, models, inference engine)

## Estimated Effort

**Time Estimate:** 2-3 days
**Complexity:** Medium
**Priority:** High (foundational component)

## Additional Notes

This configuration management system is designed to be production-ready and follows industry best practices for:
- Separation of configuration from code
- Environment-specific deployments
- Parameter validation and error handling
- Maintainability and extensibility

The implementation should support future enhancements such as:
- Dynamic configuration reloading
- Configuration versioning
- A/B testing parameter variations
- Integration with external configuration management systems
